main/src/ca/uwaterloo/flix/language/errors/EntryPointError.scala

```diff
@@ -0,0 +1,111 @@
+import ca.uwaterloo.flix.util.Formatter
+
+sealed trait EntryPointError extends CompilationMessage {
+  val kind: String = "Entry Point Error"
+}
+
+object EntryPointError {
+
+  private implicit val audience: Audience = Audience.External
+
+  /**
+    * Error indicating an unexpected argument type to an entry point function.
+    *
+    * @param sym the entry point function.
+    * @param tpe the argument type.
+    * @param loc the location where the error occurred.
+    */
+  case class UnexpectedEntryPointArg(sym: Symbol.DefnSym, tpe: Type, loc: SourceLocation) extends EntryPointError {
+
+    override def summary: String = s"Unexpected entry point argument type: ${FormatType.formatWellKindedType(tpe)}."
+
+    override def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> The argument type: '${red(FormatType.formatWellKindedType(tpe))}' is not a valid entry point argument type.
+         |
+         |${code(loc, "unexpected entry point argument type.")}
+         |
+         |The argument type must be one of:
+         |  - ${FormatType.formatWellKindedType(Type.mkArray(Type.Str, SourceLocation.Unknown))}
+         |  - ${FormatType.formatWellKindedType(Type.Unit)}
+         |""".stripMargin
+    }
+
+    override def explain(formatter: Formatter): Option[String] = None
+  }
+
+  /**
+    * Error indicating an unexpected result type to an entry point function.
+    *
+    * @param sym the entry point function.
+    * @param tpe the result type.
+    * @param loc the location where the error occurred.
+    */
+  case class UnexpectedEntryPointResult(sym: Symbol.DefnSym, tpe: Type, loc: SourceLocation) extends EntryPointError {
+
+    override def summary: String = s"Unexpected entry point result type: ${FormatType.formatWellKindedType(tpe)}."
+
+    override def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> The result type: '${red(FormatType.formatWellKindedType(tpe))}' is not a valid entry point result type.
+         |
+         |${code(loc, "unexpected entry point result type.")}
+         |
+         |The result type must be one of:
+         |  - ${FormatType.formatWellKindedType(Type.Unit)}
+         |  - a type with a ToString instance
+         |""".stripMargin
+    }
+
+    override def explain(formatter: Formatter): Option[String] = None
+  }
+
+  /**
+    * Error indicating too many arguments to an entry point function.
+    *
+    * @param sym the entry point function.
+    * @param loc the location where the error occurred.
+    */
+  case class TooManyEntryPointArgs(sym: Symbol.DefnSym, loc: SourceLocation) extends EntryPointError {
+
+    override def summary: String = s"Too many entry point arguments."
+
+    override def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>>
```

Comment:
Missing some text.

How about: "Expected 0 or 1 argument for main. Found XYZ."

or similar phrasing like above.